includes/sanitizers/class-amp-img-sanitizer.php

```diff
@@ -81,13 +85,52 @@ public function get_selector_conversion_mapping() {
 		];
 	}
 
+	/**
+	 * Convert picture element into image element or mark as px verified.
+	 *
+	 * @return void
+	 */
+	protected function process_picture_elements() {
+
+		$picture_img_query = $this->dom->xpath->query( '//picture/img' );
+		if ( 0 === $picture_img_query->length ) {
+			return;
+		}
+
+		/** @var Element $img_element */
+		foreach ( $picture_img_query as $img_element ) {
+			/** @var Element $picture_element */
+			$picture_element = $img_element->parentNode;
+
+			if ( true === $this->args['allow_picture'] ) {
+				ValidationExemption::mark_node_as_px_verified( $picture_element );
+				foreach ( $picture_element->getElementsByTagName( Tag::SOURCE ) as $source_element ) {
+					ValidationExemption::mark_node_as_px_verified( $source_element );
+
+					// Mark width/height attributes as PX-verified as well since they aren't known yet in the validator. See <https://github.com/whatwg/html/pull/5894>.
+					foreach ( [ Attribute::WIDTH, Attribute::HEIGHT ] as $dimension_attr ) {
+						$attr_node = $source_element->getAttributeNode( $dimension_attr );
+						if ( $attr_node instanceof DOMAttr ) {
+							ValidationExemption::mark_node_as_px_verified( $attr_node );
+						}
+					}
+				}
+			} else {
+				$picture_element->removeChild( $img_element );
+				$picture_element->parentNode->replaceChild( $img_element, $picture_element );
```

Comment:
I don't have much experience with the DOM API in PHP but is it necessary to remove the IMG tag first from the PICTURE and then replace the entire PICTURE with an IMG? Wouldn't just the `replaceChild()` suffice?

Reply:
I recall this has been necessary since replacing a parent with a child while the child is still in the DOM could cause an error, if I remember correctly.